optuna_dashboard/_app.py

```diff
@@ -411,6 +412,45 @@ def save_study_note(study_id: int) -> BottleViewReturn:
         response.status = 204  # No content
         return {}
 
+    @app.post("/api/studies/<study_id:int>/<trial_id:int>/tell")
+    @json_api_view
+    def tell_trial(study_id: int, trial_id: int) -> BottleViewReturn:
+        s = request.json.get("state", None)
+        vs = request.json.get("values", None)
+
+        try:
+            values = [float(v) for v in vs] if vs is not None else vs
+        except ValueError:
+            response.status = 400  # Bad request
+            return {"reason": "You need to pass float castable values"}
```

Comment:
Please return 400 response when `state == "Complete" and values is None` as well.

Reply:
Fixed